TechnicalDocuments/Proposals/Feedback Navigation.md

```diff
@@ -0,0 +1,81 @@
+# Feedback Navigation
+
+* Author(s): Sergey Shulga
+* Review Manager: TBD
+
+## Introduction
+
+As all of you probably know in our codebase we use `Route` based navigation, where VM has a signal of routes which emits a specific route and FlowController just listens to it and performs navigation.
+
+Although the current concept is fine and we've been happy with it for a while now, it does look a bit weird when it comes to integration with RAF.
+For example, we've recently agreed that we derive all routes from the `State`  of the VM. Also, sometimes we need to pass closures to the routes as we need to notify our VM when a user selects something on the child screen.
```

Comment:
Not just selects, but also cancels action. In this case parent view model should be restored to the state that was before child view was triggered. Otherwise it will be in inconsistent state and it might not be possible to make selection again (depends on the reducer implementation).